### PR TITLE
arrays/period: allow parsing of PeriodDtype columns from read_csv

### LINKED ISSUE #26934: Issue with parsing PeriodDtype columns in read_csv()
#### Code Sample

```python
df = pd.DataFrame({'Int': [1, 2, 3], 'Period': pd.period_range(start="2019-01", end="2019-03", freq="M")})
df.to_csv("PeriodDtype.csv")
pd.read_csv("PeriodDtype.csv", dtype={"Int": np.int64, "Period": pd.PeriodDtype("M")})

```
#### Problem description

Using pandas 0.24.2, I wrote a simple data frame with the following dtypes into a csv file,
```
Int           int64
Period    period[M]
dtype: object
```
When I tried to read it back in, I found that `read_csv()` could not parse `PeriodDtype("M")`. I got the following error message:
```
NotImplementedError: Extension Array: <class 'pandas.core.arrays.period.PeriodArray'> 
must implement _from_sequence_of_strings in order to be used in parser methods
```
I saw a similar issue #24542 raised for Datetime dtype. It seems that `_from_sequence_of_strings()` is also not defined for `PeriodArray`, which prevents parsing columns with `PeriodDtype`.

I think adding `_from_sequence_of_strings()` for `PeriodArray` would be a good enhancement. If that is the case I would be interested in making that change.

#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here below this line]

INSTALLED VERSIONS
------------------
commit: None
python: 3.7.2.final.0
python-bits: 64
OS: Darwin
OS-release: 18.6.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.24.2
pytest: 3.8.0
pip: 10.0.1
setuptools: 40.2.0
Cython: 0.28.5
numpy: 1.16.4
scipy: 1.1.0
pyarrow: None
xarray: None
IPython: 6.5.0
sphinx: 1.7.9
patsy: 0.5.0
dateutil: 2.7.3
pytz: 2018.5
blosc: None
bottleneck: 1.2.1
tables: 3.4.4
numexpr: 2.6.8
feather: None
matplotlib: 2.2.3
openpyxl: 2.5.6
xlrd: 1.1.0
xlwt: 1.3.0
xlsxwriter: 1.1.0
lxml.etree: 4.2.5
bs4: 4.6.3
html5lib: 1.0.1
sqlalchemy: 1.2.11
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
gcsfs: None

</details>

### PR DESCRIPTION
**Pull request by chibby0ne**
Fixes: https://github.com/pandas-dev/pandas/issues/26934

Signed-off-by: Antonio Gutierrez <chibby0ne@gmail.com>

- [x] closes #xxxx
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Review by jreback — CHANGES_REQUESTED** [review]
tests

**Comment by chibby0ne** [comment]
> tests

Done! Thank you for the swift reply ;)!

File: pandas/core/arrays/period.py [hunk 1]
```diff
@@ -253,6 +253,10 @@ def _from_sequence(
         ordinals = libperiod.extract_ordinals(periods, freq)
         return cls(ordinals, freq=freq)
 
+    @classmethod
+    def _from_sequence_of_strings(cls, strings, dtype=None, copy=False):
+        return cls._from_sequence(strings, dtype, copy)
```
Comment (jbrockmendel) [inline pandas/core/arrays/period.py]: Why does this need a new method?

Comment (chibby0ne) [reply pandas/core/arrays/period.py]: This is the method that gets used by the parser when reading Extension Arrays. The issue explains it with a bit more context: https://github.com/pandas-dev/pandas/issues/26934

**Comment by chibby0ne** [comment]
Can someone explain how the codecov/project failed? It looks as if the coverage in all the files dropped, even though I only changed 2 files.

**Comment by jbrockmendel** [comment]
> Can someone explain how the codecov/project failed? 

Happens sometimes, seems random.  Rebase and push again and it usually fixes itself

**Comment by chibby0ne** [comment]
> > Can someone explain how the codecov/project failed?
> 
> Happens sometimes, seems random. Rebase and push again and it usually fixes itself

Weird. Thanks for the suggestion though ;)

**Review by WillAyd — APPROVED** [review]
lgtm

**Review by gfyoung — APPROVED** [review]

File: doc/source/whatsnew/v0.25.0.rst [hunk 2]
```diff
@@ -1186,6 +1186,7 @@ ExtensionArray
 - :meth:`Series.count` miscounts NA values in ExtensionArrays (:issue:`26835`)
 - Added ``Series.__array_ufunc__`` to better handle NumPy ufuncs applied to Series backed by extension arrays (:issue:`23293`).
 - Keyword argument ``deep`` has been removed from :meth:`ExtensionArray.copy` (:issue:`27083`)
+- Allow parsing of `PeriodDType` columns when using `read_csv()` (:issue:`26934`)
```
Comment (TomAugspurger) [inline doc/source/whatsnew/v0.25.0.rst]:
```suggestion
- Allow parsing of :class:`PeriodDType` columns when using `read_csv()` (:issue:`26934`)
```

**Review by TomAugspurger — APPROVED** [review]

File: doc/source/whatsnew/v0.25.0.rst [hunk 3]
```diff
@@ -1186,6 +1186,7 @@ ExtensionArray
 - :meth:`Series.count` miscounts NA values in ExtensionArrays (:issue:`26835`)
 - Added ``Series.__array_ufunc__`` to better handle NumPy ufuncs applied to Series backed by extension arrays (:issue:`23293`).
 - Keyword argument ``deep`` has been removed from :meth:`ExtensionArray.copy` (:issue:`27083`)
+- Allow parsing of :class:`PeriodDtype` columns when using `read_csv()` (:issue:`26934`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.25.0.rst]: this should be in the IO section

use ``:func:`read_csv` ``

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v0.25.0.rst (hunk 3 above)
Comment (chibby0ne) [reply doc/source/whatsnew/v0.25.0.rst]: Done thanks a lot for the quick feedback again!

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @chibby0ne

**Comment by chibby0ne** [comment]
> thanks @chibby0ne

Thank you for your support @jreback @TomAugspurger @jbrockmendel @WillAyd !

**Pull request merged**